src/overlays/actors/ovl_Demo_Du/z_demo_du.c

```diff
@@ -1,228 +1,1044 @@
+    Player* player = PLAYER;
+    Vec3f* projectedPos = &player->actor.projectedPos;
+
+    func_80078914(projectedPos, NA_SE_EN_DARUNIA_HIT_LINK);
+    Audio_PlaySoundGeneral(NA_SE_VO_LI_DAMAGE_S_KID, projectedPos, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+void DemoDu_CsPlaySfx_HitBreast(DemoDu* this) {
+    func_80078914(&this->actor.projectedPos, NA_SE_EN_DARUNIA_HIT_BREAST - SFX_FLAG);
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+// Sfx played when Link is escaping from the gorons at the end of the scene.
+void DemoDu_CsPlaySfx_LinkEscapeFromGorons(GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.frames == 1400) {
+        Player* player = PLAYER;
+
+        Audio_PlaySoundGeneral(NA_SE_VO_LI_FALL_L_KID, &player->actor.projectedPos, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+    }
+}
+
+// Cutscene: Darunia gives Link the Goron's Ruby.
+// Sfx played when Link is surprised by Darunia falling from the sky.
+void DemoDu_CsPlaySfx_LinkSurprised(GlobalContext* globalCtx) {
+    if (globalCtx->csCtx.frames == 174) {
+        Player* player = PLAYER;
+
+        Audio_PlaySoundGeneral(NA_SE_VO_LI_SURPRISE_KID, &player->actor.projectedPos, 4U, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+    }
+}
+
+void DemoDu_CsGoronsRuby_UpdateFaceTextures(DemoDu* this, GlobalContext* globalCtx) {
+    u16* frames = &globalCtx->csCtx.frames;
+
+    if (*frames < 260) {
+        DemoDu_UpdateEyes(this);
+        DemoDu_SetMouthTexIndex(this, 0);
+    } else if (*frames < 335) {
+        DemoDu_UpdateEyes(this);
+        DemoDu_SetMouthTexIndex(this, 3);
+    } else if (*frames < 365) {
+        DemoDu_SetEyeTexIndex(this, 3);
+        DemoDu_SetMouthTexIndex(this, 1);
+    } else if (*frames < 395) {
+        DemoDu_SetEyeTexIndex(this, 0);
+        DemoDu_SetMouthTexIndex(this, 3);
+    } else if (*frames < 410) {
+        DemoDu_UpdateEyes(this);
+        DemoDu_SetMouthTexIndex(this, 0);
+    } else {
+        DemoDu_UpdateEyes(this);
+        DemoDu_SetMouthTexIndex(this, 3);
+    }
+}
+
+void func_8096A630(DemoDu* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f pos = this->actor.world.pos;
+
+    pos.y += kREG(5);
+    func_80033480(globalCtx, &pos, kREG(1) + 100.0f, kREG(2) + 10, kREG(3) + 300, kREG(4), 0);
+    DemoDu_CsPlaySfx_GoronLanding(this);
+}
+
+void DemoDu_CsGoronsRuby_SpawnDustWhenHittingLink(DemoDu* this, GlobalContext* globalCtx) {
+    static Vec3f sDustPosOffsets[] = { { 11.0f, -11.0f, -6.0f }, { 0.0f, 14.0f, -13.0f }, { 14.0f, -2.0f, -10.0f },
+                                       { 10.0f, -6.0f, -8.0f },  { 8.0f, 6.0f, 8.0f },    { 13.0f, 8.0f, -10.0f },
+                                       { -14.0f, 1.0f, -14.0f }, { 5.0f, 12.0f, -9.0f },  { 11.0f, 6.0f, -7.0f },
+                                       { 14.0f, 14.0f, -14.0f } };
```

Comment:
id say these should go outside the function, unless this is needed to match. also add a trailing comma and reformat

Reply:
To move this out of the function, I would need to move the others variables that are being initialized out (`velocity`, `accel`, `primColor` and `envColor`)

Reply:
hm, ok. I probably would personally, but its a tough call either way.
for now put a comma at the end of the list so its formatted more nicely, and others can weigh in if they also think it should move out